### PR TITLE
fix(popover): clamp the overlay horizontally so a wide panel stays on-screen

### PR DESCRIPTION
**Pull request by anilcancakir**
## What

`WPopover` now folds a corrective horizontal offset into the `CompositedTransformFollower` when the overlay would otherwise overflow the viewport, so a wide panel opened near a screen edge stays fully on-screen (seated 8px from the edge) instead of clipping off the right side.

## Why

On narrow (mobile) viewports a notification popup anchored near the right edge opened off-screen and was effectively invisible. The clamp only engages on genuine off-screen overflow, so centered and left-anchored popovers are unaffected.

## Changes

- `lib/src/widgets/w_popover.dart`: `computeHorizontalClamp(...)` static helper plus a top-level wrapper that adjusts the follower offset only when the panel would overflow.
- `test/widgets/w_popover_test.dart`: coverage for the clamp (overflow corrected, no-overflow untouched).

## Verification

`dart analyze` clean, `dart format` clean, `flutter test` green.

## Summary by CodeRabbit

* **New Features**
  * Popovers now stay within the screen horizontally, even in edge cases where automatic flipping was already applied.
  * Added a new public helper for calculating horizontal clamping behavior.

* **Bug Fixes**
  * Improved popover positioning so wide panels no longer overflow past the left or right edges of the viewport.
  * Better handling for centered and edge-anchored popovers to keep content visible with consistent margins.

### CODE REVIEW

File: lib/src/widgets/w_popover.dart [hunk 1]
```diff
@@ -342,6 +367,56 @@ class _WPopoverState extends State<WPopover> {
...
+    double correction = 0;
+    if (followerRight > screenSize.width) {
+      correction = (screenSize.width - margin) - followerRight;
+    }
+    // Re-check the left edge after any right-edge pull. The left clamp wins so a
+    // popover wider than the viewport keeps its start (top-left) on-screen.
+    if (followerLeft + correction < 0) {
+      correction = margin - followerLeft;
+    }
+    return correction;
```
[the remark this answers is not in the document]
Comment (anilcancakir) [reply lib/src/widgets/w_popover.dart:429]: Fixed in 535e375. The left-edge clamp is now capped so it never shoves the right edge back off-screen: when the panel is too wide for both 8px margins it seats flush to the viewport edge (giving up the tighter margin) instead of re-overflowing. Added a regression unit test for the popoverWidth in (screenWidth - margin, screenWidth] case.

File: lib/src/widgets/w_popover.dart [hunk 2]
```diff
@@ -700,10 +775,32 @@ class _WPopoverState extends State<WPopover> {
...
-    final effectiveOffset = isTopAlignment
+    final Offset baseOffset = isTopAlignment
         ? Offset(widget.offset.dx, -widget.offset.dy)
         : widget.offset;
 
+    // Horizontal viewport clamp: autoFlip only swaps the anchored side, so a
+    // panel wider than the space on both sides still spills off-screen after the
+    // flip (a wide popover next to a near-centered trigger on a narrow phone).
+    // Pull it back inside the viewport using the same width estimate the overlay
+    // renders. Gated on autoFlip: that flag already promises "keep me on screen".
```
[the remark this answers is not in the document]
Comment (anilcancakir) [reply lib/src/widgets/w_popover.dart:798]: Fixed in 535e375. Added a `[Unreleased]` -> `Fixed` bullet in `CHANGELOG.md` recording the WPopover horizontal clamp.

File: test/widgets/w_popover_test.dart [hunk 3]
```diff
@@ -838,7 +838,151 @@ void main() {
...
+          screenSize: const Size(375, 800),
+          offset: Offset.zero,
+        );
+
+        expect(result, closeTo(-153, 0.001));
+      });
+    });
+
     group('Auto-Flip Widget Integration', () {
+      testWidgets(
```
[the remark this answers is not in the document]
Comment (anilcancakir) [reply test/widgets/w_popover_test.dart:964]: Fixed in f6a4184. Added `setUp(WindParser.clearCache)` at the top of `main()` so every group clears the parser style cache between tests.

File: test/widgets/w_popover_test.dart [hunk 4]
```diff
@@ -838,7 +838,151 @@ void main() {
...
+          triggerSize: const Size(50, 30),
+          popoverWidth: 500,
+          screenSize: const Size(400, 300),
+          offset: Offset.zero,
+        );
+
+        expect(result, closeTo(-92, 0.001));
+      });
+
+      test('center-anchored popover overflowing right is pulled in', () {
```
[the remark this answers is not in the document]
Comment (anilcancakir) [reply test/widgets/w_popover_test.dart:923]: Fixed in f6a4184. Added a unit test for a panel too wide for both margins (popoverWidth 396 on a 400px viewport): it asserts the correction is -26 and the right edge lands flush at 400 rather than being pushed past it. This test fails on the old clamp ordering.

File: lib/src/widgets/w_popover.dart [hunk 5]
```diff
@@ -700,10 +775,32 @@ class _WPopoverState extends State<WPopover> {
...
+        triggerPosition: triggerBox.localToGlobal(Offset.zero),
+        triggerSize: triggerBox.size,
+        popoverWidth: popoverWidth,
+        screenSize: MediaQuery.sizeOf(context),
+        offset: baseOffset,
+      );
+      if (dxCorrection != 0) {
+        effectiveOffset = Offset(baseOffset.dx + dxCorrection, baseOffset.dy);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (anilcancakir) [reply lib/src/widgets/w_popover.dart:814]: Verified, intentionally not changed. The rendered width of a flexible-width popover is not known at offset-computation time (content lays out after positioning). The suggested upper bound `effectiveMaxWidth` defaults to the screen width (see #127), so using it here would make the clamp treat every flexible popover as screen-wide and pull it to the left margin, regressing the common menu case. For fixed-width popovers (`fixedWidth`) the estimate is already exact. The `triggerWidth` heuristic plus the now-corrected clamp (which no longer pushes an edge off-screen for the estimated width) is the pragmatic choice; measuring the true rendered width would need a post-layout pass and is out of scope for this fix.

**Comment by anilcancakir** [comment]
@coderabbitai review
